### ISSUE TITLE
Segmentation fault (Core dumped)

### ISSUE THREAD

**Issue by deepkshikha** [KleinYuan/easy-yolo#5]
I am getting This error After running make train. I have created the folder according to as it is described . My Images are in  .JPG format  and verying size. 

Does we Require to change anything in src folder as it is not mentioned here to make a change I am curious since yolo.c took no of class and class name and annotation path If these are needed to change. Or because of that error is occuring

**Comment by deepkshikha** [comment]
ubuntu@ip-172-30-6-196:~/easy-yolo$ make train
/home/ubuntu/easy-yolo/darknet detector train cfg/easy.data cfg/easy.cfg darknet                                                                                        19_448.conv.23
easy
layer     filters    size              input                output
    0 conv     16  3 x 3 / 1   448 x 448 x   3   ->   448 x 448 x  16
    1 max          2 x 2 / 2   448 x 448 x  16   ->   224 x 224 x  16
    2 conv     32  3 x 3 / 1   224 x 224 x  16   ->   224 x 224 x  32
    3 max          2 x 2 / 2   224 x 224 x  32   ->   112 x 112 x  32
    4 conv     64  3 x 3 / 1   112 x 112 x  32   ->   112 x 112 x  64
    5 max          2 x 2 / 2   112 x 112 x  64   ->    56 x  56 x  64
    6 conv    128  3 x 3 / 1    56 x  56 x  64   ->    56 x  56 x 128
    7 max          2 x 2 / 2    56 x  56 x 128   ->    28 x  28 x 128
    8 conv    256  3 x 3 / 1    28 x  28 x 128   ->    28 x  28 x 256
    9 max          2 x 2 / 2    28 x  28 x 256   ->    14 x  14 x 256
   10 conv    448  3 x 3 / 1    14 x  14 x 256   ->    14 x  14 x 448
   11 max          2 x 2 / 1    14 x  14 x 448   ->    14 x  14 x 448
   12 conv   1024  3 x 3 / 1    14 x  14 x 448   ->    14 x  14 x1024
   13 conv   1024  3 x 3 / 1    14 x  14 x1024   ->    14 x  14 x1024
   14 conv     30  1 x 1 / 1    14 x  14 x1024   ->    14 x  14 x  30
   15 detection
Loading weights from darknet19_448.conv.23...Done!
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Resizing
448
Cannot load image "131"
Cannot load image "117"
Cannot load image "1006"
Cannot load image "1037"
Cannot load image "1028"
Cannot load image "1007"
Cannot load image "1069"
Cannot load image "1016"
Couldn't open file: 1006
Couldn't open file: 1037
Couldn't open file: 1007
make: *** [train] Segmentation fault (core dumped)
 Please suggest something

**Comment by KleinYuan** [comment]
@deepkshikha could you check/post your folder layout? It seems that the path in your txt is not correct, as the error complained.

**Comment by deepkshikha** [comment]
Thanks for Response Sure sir
ubuntu@ip-172-30-6-196:~/easy-yolo$ ls
backup  bad.list  cfg  darknet  darknet19_448.conv.23  devkit  easy.names  LICENSE  Makefile  obj  pythonWrapper.py  README.md  results  scripts  src
ubuntu@ip-172-30-6-196:~/easy-yolo$ cd devkit/
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit$ ls
2017
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017$ ls
Annotations  Images  ImageSets  Labels
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017/ImageSets$ ls
train.txt  val.txt
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017/Images$ ls
1000.png  1006.png  1011.png  1017.png  1022.png  1028.png  1033.png  1040.png  1048.png  1054.png  1060.png  106.png   108.png  1136.png   117.png  127.png  138.png
1001.png  1007.png  1012.png  1018.png  1023.png  1029.png  1034.png  1041.png  104.png   1055.png  1062.png  1070.png  109.png  113.png    118.png  128.png  139.png
1002.png  1008.png  1013.png  1019.png  1024.png  102.png   1037.png  1042.png  1050.png  1056.png  1064.png  1071.png  10.png   114.png    119.png  129.png  13.png
1003.png  1009.png  1014.png  101.png   1025.png  1030.png  1038.png  1044.png  1051.png  1058.png  1066.png  1073.png  110.png  115.png    11.png   130.png  670.png
1004.png  100.png   1015.png  1020.png  1026.png  1031.png  1039.png  1045.png  1052.png  1059.png  1067.png  1074.png  111.png  116.png    120.png  131.png
1005.png  1010.png  1016.png  1021.png  1027.png  1032.png  103.png   1047.png  1053.png  105.png   1069.png  107.png   112.png  11751.png  121.png  137.png
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017/Images$ cd ..
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017$ cd Annotations/
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017/Annotations$ ls
1000.xml  1006.xml  1011.xml  1017.xml  1022.xml  1028.xml  1033.xml  1040.xml  1048.xml  1054.xml  1060.xml  106.xml   108.xml  1136.xml   117.xml  127.xml  138.xml
1001.xml  1007.xml  1012.xml  1018.xml  1023.xml  1029.xml  1034.xml  1041.xml  104.xml   1055.xml  1062.xml  1070.xml  109.xml  113.xml    118.xml  128.xml  139.xml
1002.xml  1008.xml  1013.xml  1019.xml  1024.xml  102.xml   1037.xml  1042.xml  1050.xml  1056.xml  1064.xml  1071.xml  10.xml   114.xml    119.xml  129.xml  13.xml
1003.xml  1009.xml  1014.xml  101.xml   1025.xml  1030.xml  1038.xml  1044.xml  1051.xml  1058.xml  1066.xml  1073.xml  110.xml  115.xml    11.xml   130.xml  670.xml
1004.xml  100.xml   1015.xml  1020.xml  1026.xml  1031.xml  1039.xml  1045.xml  1052.xml  1059.xml  1067.xml  1074.xml  111.xml  116.xml    120.xml  131.xml
1005.xml  1010.xml  1016.xml  1021.xml  1027.xml  1032.xml  103.xml   1047.xml  1053.xml  105.xml   1069.xml  107.xml   112.xml  11751.xml  121.xml  137.xml
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017/Annotations$ cd ..
ubuntu@ip-172-30-6-196:~/easy-yolo/devkit/2017$ cd Labels/

**Comment by deepkshikha** [comment]
Easy-Yolo
  cfg
  devkit
      2017
           Images
           Annotations
           ImageSet
           Labels
  scripts
  src
  easy.names
  makeFile

**Comment by deepkshikha** [comment]
![2](https://user-images.githubusercontent.com/13184577/33040140-2de65556-ce60-11e7-85eb-9f7a6e97afc3.jpeg)
![1](https://user-images.githubusercontent.com/13184577/33040142-2e23d8d6-ce60-11e7-9f5d-f9c839c28e79.jpeg)
![3](https://user-images.githubusercontent.com/13184577/33040143-2e602228-ce60-11e7-81b5-a1f357a964a4.jpeg)
![4](https://user-images.githubusercontent.com/13184577/33040144-2e9d7dee-ce60-11e7-9a2b-21efba12d55b.jpeg)
![5](https://user-images.githubusercontent.com/13184577/33040145-2eda5a48-ce60-11e7-86a1-34ea4913859b.jpeg)

**Comment by deepkshikha** [comment]
![2](https://user-images.githubusercontent.com/13184577/33040219-65a82370-ce60-11e7-9d9f-cbbf4c132c50.jpeg)
![1](https://user-images.githubusercontent.com/13184577/33040220-65fb00ae-ce60-11e7-9221-d74410afa4ca.jpeg)
![3](https://user-images.githubusercontent.com/13184577/33040223-663eec1a-ce60-11e7-8dd9-e0a5a9cbd3ee.jpeg)
![4](https://user-images.githubusercontent.com/13184577/33040224-667dd394-ce60-11e7-9547-71ffe715d83e.jpeg)
![5](https://user-images.githubusercontent.com/13184577/33040225-66bd131a-ce60-11e7-801d-b3e32e67a47b.jpeg)

**Comment by deepkshikha** [comment]
![1](https://user-images.githubusercontent.com/13184577/33040254-7d79779c-ce60-11e7-95af-649cec2d5d8b.jpeg)

**Comment by deepkshikha** [comment]
Sure sir Below are the folder arrangements

![1](https://user-images.githubusercontent.com/13184577/33040283-9d373eca-ce60-11e7-8265-86f93ec72460.JPG)
![2](https://user-images.githubusercontent.com/13184577/33040284-9d753dce-ce60-11e7-869b-ff17ca8dd4aa.JPG)
![3](https://user-images.githubusercontent.com/13184577/33040285-9db2cb80-ce60-11e7-882b-3727e34b7ec6.JPG)
![4](https://user-images.githubusercontent.com/13184577/33040286-9def328c-ce60-11e7-94a6-79ef19c1a820.JPG)
![5](https://user-images.githubusercontent.com/13184577/33040287-9e35fbcc-ce60-11e7-85d4-4ba7f3c10e0b.JPG)

**Comment by deepkshikha** [comment]
Sure Sir
![folderarrangements](https://user-images.githubusercontent.com/13184577/33040470-31189d64-ce61-11e7-9932-bd40a689502a.JPG)

**Comment by deepkshikha** [comment]
Do I need to any change in the below part of yolo.c


void test_yolo(char *cfgfile, char *weightfile, char *filename, float thresh)
{
    image **alphabet = load_alphabet();
    network net = parse_network_cfg(cfgfile);
    if(weightfile){
        load_weights(&net, weightfile);
    }
    detection_layer l = net.layers[net.n-1];
    set_batch_network(&net, 1);
    srand(2222222);
    clock_t time;
    char buff[256];
    char *input = buff;
    int j;
    float nms=.4;
    box *boxes = calloc(l.side*l.side*l.n, sizeof(box));
    float **probs = calloc(l.side*l.side*l.n, sizeof(float *));
    for(j = 0; j < l.side*l.side*l.n; ++j) probs[j] = calloc(l.classes, sizeof(float *));
    while(1){
        if(filename){
            strncpy(input, filename, 256);
        } else {
            printf("Enter Image Path: ");
            fflush(stdout);
            input = fgets(input, 256, stdin);
            if(!input) return;
            strtok(input, "\n");
        }
        image im = load_image_color(input,0,0);
        image sized = resize_image(im, net.w, net.h);
        float *X = sized.data;
        time=clock();
        network_predict(net, X);
        printf("%s: Predicted in %f seconds.\n", input, sec(clock()-time));
        get_detection_boxes(l, 1, 1, thresh, probs, boxes, 0);
        if (nms) do_nms_sort(boxes, probs, l.side*l.side*l.n, l.classes, nms);
        //draw_detections(im, l.side*l.side*l.n, thresh, boxes, probs, voc_names, alphabet, 20);
        draw_detections(im, l.side*l.side*l.n, thresh, boxes, probs, voc_names, alphabet, 20);
        save_image(im, "predictions");
        show_image(im, "predictions");

        free_image(im);
        free_image(sized);
#ifdef OPENCV
        cvWaitKey(0);
        cvDestroyAllWindows();
#endif
        if (filename) break;
    }
}

void run_yolo(int argc, char **argv)
{
    char *prefix = find_char_arg(argc, argv, "-prefix", 0);
    float thresh = find_float_arg(argc, argv, "-thresh", .2);
    int cam_index = find_int_arg(argc, argv, "-c", 0);
    int frame_skip = find_int_arg(argc, argv, "-s", 0);
    if(argc < 4){
        fprintf(stderr, "usage: %s %s [train/test/valid] [cfg] [weights (optional)]\n", argv[0], argv[1]);
        return;
    }

    char *cfg = argv[3];
    char *weights = (argc > 4) ? argv[4] : 0;
    char *filename = (argc > 5) ? argv[5]: 0;
    if(0==strcmp(argv[2], "test")) test_yolo(cfg, weights, filename, thresh);
    else if(0==strcmp(argv[2], "train")) train_yolo(cfg, weights);
    else if(0==strcmp(argv[2], "valid")) validate_yolo(cfg, weights);
    else if(0==strcmp(argv[2], "recall")) validate_yolo_recall(cfg, weights);
    else if(0==strcmp(argv[2], "demo")) demo(cfg, weights, thresh, cam_index, filename, voc_names, 20, frame_skip, prefix, .5);
}

**Comment by deepkshikha** [comment]
I made the changes only this part of yolo.c


#include "network.h"
#include "detection_layer.h"
#include "cost_layer.h"
#include "utils.h"
#include "parser.h"
#include "box.h"
#include "demo.h"

#ifdef OPENCV
#include "opencv2/highgui/highgui_c.h"
#endif

char *voc_names[] = {"crater"};

void train_yolo(char *cfgfile, char *weightfile)
{
    char *train_images = "/home/ubuntu/easy-yolo/devkit/2017/ImageSets/train.txt";
    char *backup_directory = "/home/ubuntu/easy-yolo/backup/";
    srand(time(0));
    char *base = basecfg(cfgfile);
    printf("%s\n", base);
    float avg_loss = -1;
    network net = parse_network_cfg(cfgfile);
    if(weightfile){
        load_weights(&net, weightfile);
    }

**Comment by KleinYuan** [comment]
@deepkshikha man, I cannot really know what you have changed by you showing code in this way.
But obviously, your error is throw from [here](https://github.com/KleinYuan/easy-yolo/blob/837863090e465ee0bbe43760bbbd4b1f2efd2cf1/src/image.c#L489). You can try to add some logs here to see specifically why it failed. Should be fairly easy.

**Comment by deepkshikha** [comment]
I just changed two lines in yolo.c
the path of training set
char *train_images = "/home/ubuntu/easy-yolo/devkit/2017/ImageSets/train.txt";
char *backup_directory = "/home/ubuntu/easy-yolo/backup/";


Do I need to change anything else in yolo.c ??

**Comment by deepkshikha** [comment]
Thanks It works There is some error in .txt file

**Comment by KleinYuan** [comment]
@deepkshikha  Neat!

**Comment by deepkshikha** [comment]
Training is going on But I have a confusion don't we need to change paths in yolo.c ?

**Comment by KleinYuan** [comment]
@deepkshikha That's just a default path I believe. Eventually, it will load the path from the config/data file

**Comment by deepkshikha** [comment]
Thanks a lot .What's the expected time for training ? I am using AWS gpu compute instance .

**Comment by KleinYuan** [comment]
@deepkshikha it's up to how many images you are feeding in.

**Comment by deepkshikha** [comment]
Is there some resource from where I can read how this is working . That is apart from creating folders in this manner why we created and rest of the things

**Comment by KleinYuan** [comment]
@deepkshikha 
Donno how to recommend then. Read Ian Goodfellow's <<Deep Learning>> maybe?

**Comment by deepkshikha** [comment]
Thanks
